scala/scala.bzl

```diff
@@ -446,39 +464,42 @@ def _scala_binary_impl(ctx):
   cjars += [ctx.file._scalareflect]
   rjars += [ctx.outputs.jar, ctx.file._scalalib, ctx.file._scalareflect]
   rjars += _collect_jars(ctx.attr.runtime_deps).runtime
-  _write_launcher(ctx, rjars)
+  _write_launcher(
+      ctx = ctx,
+      rjars = rjars,
+      main_class = ctx.attr.main_class,
+      jvm_flags = "",
+      args = "",
+  )
   return _scala_binary_common(ctx, cjars, rjars)
 
 def _scala_repl_impl(ctx):
   jars = _collect_jars(ctx.attr.deps)
   rjars = jars.runtime
-  rjars += [ctx.file._scalalib, ctx.file._scalareflect]
+  rjars += [ctx.file._scalalib, ctx.file._scalareflect, ctx.file._scalacompiler]
   rjars += _collect_jars(ctx.attr.runtime_deps).runtime
-  classpath = ':'.join(["$0.runfiles/%s/%s" % (ctx.workspace_name, f.short_path) for f in rjars])
-  content = """#!/bin/bash
-env JAVACMD=$0.runfiles/{repo}/{java} $0.runfiles/{repo}/{scala} {jvm_flags} -classpath {classpath} {scala_opts} "$@"
-""".format(
-    java=ctx.file._java.short_path,
-    repo=ctx.workspace_name,
-    jvm_flags=" ".join(["-J" + flag for flag in ctx.attr.jvm_flags]),
-    scala=ctx.file._scala.short_path,
-    classpath=classpath,
-    scala_opts=" ".join(ctx.attr.scalacopts),
+
+  jvm_flags = " ".join(
+      ["-Dscala.usejavacp=true"] +
+      ["-J" + flag for flag in ctx.attr.jvm_flags]
```

Comment:
I don't *think* we need `-J` if you are not passing to the scala script. We should have a way to test flags.